Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ServerGameNetPortal.cs

```diff
@@ -153,75 +154,75 @@ private void ApprovalCheck(byte[] connectionData, ulong clientId, NetworkManager
                 return;
             }
 
-            ConnectStatus gameReturnStatus;
+            var payload = System.Text.Encoding.UTF8.GetString(connectionData);
+            var connectionPayload = JsonUtility.FromJson<ConnectionPayload>(payload); // https://docs.unity3d.com/2020.2/Documentation/Manual/JSONSerialization.html

```

Comment:
A small change introduced in this PR is that the connection data is deserialized earlier, which means it now happens in the case where the game is at maximum capacity while it did not before.
While profiling, I measured the time it takes for that to be less than 0.3 ms in the editor on my machine with deep profile activated, which is small but not negligible.
I chose to leave it that way anyways because I feel it makes the code simpler and easier to follow.

Reply:
is this comment still up to date? looking at current implementation, looks like the server full check is still pretty early

Reply:
yes. The main difference here is that before that, we would do the IsServerFull check before deserializing the connecting data. Although, since we now also verify the player capacity in Lobby before starting the connection, it should not really matter.

Reply:
If you are in agreement with it, I will remove the comment in the code about it at the same time

Reply:
Tested again to be sure and it still works fine